vendor-extra/ViewsBundle/src/Event/CreateViewEvent.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ViewsBundle\Event;
+
+use FluentDOM\DOM\Element;
+use Libero\ViewsBundle\Views\View;
+use Symfony\Component\EventDispatcher\Event;
+
+final class CreateViewEvent extends Event
```

Comment:
Needs a better name.

Reply:
`ConvertView`?

Reply:
`GetViewForElement`? `ElementToView`?

Reply:
Verbs are usually out of place wrt nouns; we also don't have a system metaphor (a real world system that we could borrow naming from). `ViewCreationFromElement`?

Reply:
It doesn't create the view because it is passed in the constructor, though it may replace it later?

Reply:
Correct. Other options are:

- Pass in the arguments for the `View` and create it there.
- Manipulate the arguments for the `View` so it's only created on `getView()`(/`toView()`). Means this event is mutable, which fits as it's actually the mediator pattern.

Reply:
Is this a [`ViewBuilder`](https://www.oodesign.com/builder-pattern.html) then?

Reply:
👍  `BuildViewEvent`?

Reply:
(something that indicates it's a `Builder` pattern, if it is so)